src/backend/distributed/commands/multi_copy.c

```diff
@@ -465,7 +465,7 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 	 * PG_CATCH. Otherwise, it may be undefined in the PG_CATCH (see sigsetjmp
 	 * documentation).
 	 */
-	shardConnectionHash = CreateShardConnectionHash();
+	copyConnectionHash = CreateShardConnectionHash(MessageContext);
```

Comment:
Yeah, the surprising bit is that `hash_create` under no circumstances defaults to `CurrentMemoryContext`. I'll use that.